server/src/grpc/meta_event_service/mod.rs

```diff
@@ -259,10 +295,14 @@ async fn do_open_shard(ctx: HandlerContext, shard_info: ShardInfo) -> Result<()>
             msg: "fail to open shards in cluster",
         })?;
 
+    // Lock the shard in local, and then recover it.
+    let _guard = shard.serializing_lock.lock().await;
```

Comment:
I guess we can define a struct called `ShardSerialExec` which holding the `serial_lock` guard in its inner. And all the update operations on the shard are defined on the `ShardSerialExec`. After this, the user won't be able to operate the shard without holding the shard's `serial_lock` anymore.

Reply:
Has refactor all related logic.